### PR TITLE
Filterx isset changes bazsi

### PR DESCRIPTION
**Pull request by bazsi**
This is an alternative to #4879 that sits on top of my previous branch #4886

@alltilla

### CODE REVIEW

File: lib/filterx/object-json-array.c [hunk 1]
```diff
@@ -172,6 +172,25 @@ _set_subscript(FilterXList *s, guint64 index, FilterXObject *new_value)
...
 
+static gboolean
+_unset_index(FilterXList *s, guint64 index)
+{
+  FilterXJsonArray *self = (FilterXJsonArray *) s;
+
+  if (G_UNLIKELY(index >= JSON_ARRAY_MAX_SIZE))
+    return FALSE;
+
+  self->super.super.modified_in_place = TRUE;
```
Comment (MrAnno) [inline lib/filterx/object-json-array.c:183]: A `json_object_array_del_idx()` seems to be missing from here.

File: modules/grpc/otel/filterx/object-otel-kvlist.cpp [hunk 2]
```diff
@@ -170,6 +170,30 @@ KVList::has_subscript(FilterXObject *key) const
...
+    {
+      KeyValue &possible_kv = repeated_kv->at(i);
+      if (possible_kv.key().compare(key_c_str) == 0)
+        {
+          repeated_kv->DeleteSubrange(i, 1);
+          return true;
+        }
+    }
+
+  return true;
```
Comment (MrAnno) [inline modules/grpc/otel/filterx/object-otel-kvlist.cpp:194]: Why do we return true on this path? We couldn't unset anything.

File: lib/filterx/object-json-array.c (hunk 1 above)
Comment (bazsi) [reply lib/filterx/object-json-array.c:183]: This is a rebase issue, thanks for noticing.

File: modules/grpc/otel/filterx/object-otel-kvlist.cpp (hunk 2 above)
Comment (bazsi) [reply modules/grpc/otel/filterx/object-otel-kvlist.cpp:194]: We concluded that unset should succeed unless there's an error

File: lib/filterx/object-json-array.c (hunk 1 above)
Comment (bazsi) [reply lib/filterx/object-json-array.c:186]: fixed. I've also reviewed my latest iteration against the @alltilla's original branch I started from and I nothing stood out.

File: lib/filterx/expr-message-ref.c [hunk 3]
```diff
@@ -31,27 +31,39 @@ typedef struct _FilterXMessageRefExpr
...
+  FilterXObject *msg_ref = filterx_message_value_new_borrowed(value, value_len, t);
+  filterx_scope_register_variable(context->scope, self->handle, FALSE, msg_ref);
+  return msg_ref;
+}
+
+/* NOTE: unset on a variable that only exists in the LogMessage, without making the message writable */
+static void
+_whiteout_variable(FilterXMessageRefExpr *self, FilterXEvalContext *context)
+{
+  filterx_scope_register_variable(context->scope, self->handle, FALSE, NULL);
```
Comment (MrAnno) [inline lib/filterx/expr-message-ref.c:52]: I don't fully understand this part, but it looks like `filterx_scope_sync_to_message()` will do the unset in a delayed manner, which is what we want.

**Review by MrAnno — DISMISSED** [review]
There is a Light style-check problem in the CI, the macOS failures are unrelated.

LGTM otherwise.

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/filterx/expr-message-ref.c (hunk 3 above)
Comment (bazsi) [reply lib/filterx/expr-message-ref.c:52]: But only if there's an entry in the scope with a NULL value, which is what this does.

**Pull request merged**